doc/manual/src/protocols/store-path.md

```diff
@@ -0,0 +1,104 @@
+# Complete Store Path Calculation
+
+This is the complete specification for how store paths are calculated.
+
+Regular users do *not* need to know this information --- store paths can be treated as black boxes computed from the properties of the store objects they refer to.
+But for those interested in exactly how Nix works, e.g. if they are reimplementing it, this information can be useful.
+
+```bnf
+<realized-path> ::= <store-dir>/<digest>-<name>
+```
+where
+
+- `<digest>` = base-32 representation of the first 160 bits of a [SHA-256] hash of `<pre>`
```

Comment:
So you're introducing metavariable `pre` here. Can't we call that the fingerprint?

Reply:
The fingerprint and the store path syntax could be separate subsections.